src/rules/parameter-properties.ts

```diff
@@ -1,12 +1,110 @@
-import { createRule } from "../utils.js";
+import { AST_NODE_TYPES } from "@typescript-eslint/utils";
+import { isIdentifier } from "@typescript-eslint/utils/ast-utils";
+import { RuleFix } from "@typescript-eslint/utils/ts-eslint";
 
+import { createRule } from "../utils.js";
 export const rule = createRule({
 	create(context) {
 		return {
 			TSParameterProperty(node) {
+				const paramNode = node.parameter;
+				if (
+					paramNode.type !== AST_NODE_TYPES.Identifier &&
+					paramNode.type !== AST_NODE_TYPES.AssignmentPattern
+				) {
+					return;
+				}
+
+				const classBody = node.parent.parent.parent;
+				if (classBody?.type !== AST_NODE_TYPES.ClassBody) {
+					return;
+				}
+
+				const constructor = node.parent;
+				if (constructor.type !== AST_NODE_TYPES.FunctionExpression) {
+					return;
+				}
+
+				let paramName = "";
+				let typeAnnotation = "";
+				if (
+					paramNode.type === AST_NODE_TYPES.AssignmentPattern &&
+					isIdentifier(paramNode.left)
+				) {
+					paramName = paramNode.left.name;
+					typeAnnotation = context.sourceCode.getText(
+						paramNode.left.typeAnnotation,
+					);
+				} else if (paramNode.type === AST_NODE_TYPES.Identifier) {
+					paramName = paramNode.name;
+					typeAnnotation = context.sourceCode.getText(paramNode.typeAnnotation);
+				}
+
+				const modifiers: string[] = [];
+				if (node.accessibility) {
+					modifiers.push(`${node.accessibility} `);
+				}
+				if (node.readonly) {
+					modifiers.push("readonly ");
+				}
+				const fieldDeclaration = `${modifiers.join()}${paramName}${typeAnnotation};`;
+
+				const isParamAssignedInConstructor = constructor.body.body.some(
+					(node) =>
+						node.type === AST_NODE_TYPES.ExpressionStatement &&
+						node.expression.type === AST_NODE_TYPES.AssignmentExpression &&
+						node.expression.left.type === AST_NODE_TYPES.MemberExpression &&
+						node.expression.left.object.type ===
+							AST_NODE_TYPES.ThisExpression &&
+						node.expression.left.property.type === AST_NODE_TYPES.Identifier &&
+						node.expression.left.property.name === paramName,
+				);
+
 				context.report({
 					messageId: "parameterProperty",
 					node,
+					suggest: [
+						{
+							fix(fixer) {
+								const fixes: RuleFix[] = [];
+
+								fixes.push(
+									fixer.insertTextBefore(
+										classBody.body[0],
+										`${fieldDeclaration}\n`,
+									),
+								);
```

Comment:
[Style] A bit more succinct of a declaration: fixer functions can be generators! Instead of returning a full array of fixes, they can yield back each fix as they get to them. Switching over saves something like 10 lines in total 😎:

```suggestion
							*fix(fixer) {
								yield fixer.insertTextBefore(
									classBody.body[0],
									`${fieldDeclaration}\n`,
								);
```